### DISCUSSION TITLE
Spawn a new tab with 'ActivateTab'

### DISCUSSION THREAD

**Discussion by aorith** [wezterm/wezterm#1324] [Q&A] [answered]
Hi all,

Is there a way to make `ActivateTab=N` create the tab number `N` if that tab doesn't exist and switch to it if it does?

Regards,

**Comment by wez** [comment, marked as answer]
There isn't.  There are some hard to answer questions about it; what if there is only one tab and you activate tab number 100--should 99 additional tabs be created?  Feels like the answer should be "no".

**Reply by aorith** [reply]
Well I guess it's not a common workflow.
I was thinking on something similar to what can be achieved with tmux by using the following keymappings:

```
bind -N 'Select window 1' -n 'M-1' if-shell 'tmux select-window -t :1' '' 'new-window -t :1'
bind -N 'Select window 2' -n 'M-2' if-shell 'tmux select-window -t :2' '' 'new-window -t :2'
bind -N 'Select window 3' -n 'M-3' if-shell 'tmux select-window -t :3' '' 'new-window -t :3'
```

In that case, spawning tab/window 3 doesn't automatically spawn tab/window 2.

Thanks for the reply.
